src/OmnichannelChatSDK.ts

```diff
@@ -484,46 +510,91 @@ class OmnichannelChatSDK {
 
         try {
             await this.conversation!.sendFileMessage(fileMetadata, messageToSend);
+            this.scenarioMarker.completeScenario(TelemetryEvent.UploadFileAttachment);
             return messageToSend;
         } catch (error) {
             console.error(`OmnichannelChatSDK/uploadFileAttachment/error: ${error}`);
+            this.scenarioMarker.failScenario(TelemetryEvent.UploadFileAttachment);
             return error;
         }
     }
 
     public async downloadFileAttachment(fileMetadata: IFileMetadata): Promise<Blob> {
-        return this.conversation!.downloadFile(fileMetadata);
+        this.scenarioMarker.startScenario(TelemetryEvent.DownloadFileAttachment, {
+            RequestId: this.requestId,
+            ChatId: this.chatToken.chatId as string
+        })
+
+        try { 
+            const downloadedFile = this.conversation!.downloadFile(fileMetadata);
+            this.scenarioMarker.completeScenario(TelemetryEvent.DownloadFileAttachment);
+            return downloadedFile;
+        }
+        catch(e) {
+            this.scenarioMarker.failScenario(TelemetryEvent.DownloadFileAttachment);
+            return this.conversation!.downloadFile(fileMetadata);
+        }
     }
 
     public async emailLiveChatTranscript(body: IChatTranscriptBody): Promise<any> { // eslint-disable-line @typescript-eslint/no-explicit-any
         const emailTranscriptOptionalParams: IEmailTranscriptOptionalParams = {};
-        if (this.authenticatedUserToken) {
-            emailTranscriptOptionalParams.authenticatedUserToken = this.authenticatedUserToken;
+
+        this.scenarioMarker.startScenario(TelemetryEvent.EmailLiveChatTranscript, {
+            RequestId: this.requestId,
+            ChatId: this.chatToken.chatId as string
+        });
+
+        try {
+
+            if (this.authenticatedUserToken) {
+                emailTranscriptOptionalParams.authenticatedUserToken = this.authenticatedUserToken;
+            }
+            const emailRequestBody = {
+                ChatId: this.chatToken.chatId,
+                EmailAddress: body.emailAddress,
+                DefaultAttachmentMessage: body.attachmentMessage,
+                CustomerLocale: body.locale
+            };
+
+            this.scenarioMarker.completeScenario(TelemetryEvent.EmailLiveChatTranscript);
```

Comment:
I'd not complete the scenario before `OCClient.emailTranscript()` has even completed. 

I'd do:

```
const response = this.OCClient.emailTranscript(...);
this.scenarioMarker.completeScenario(TelemetryEvent.EmailLiveChatTranscript);
return response;
```